rfcs/0002-mvp.md

```diff
@@ -0,0 +1,159 @@
+
+- `postgresql://postgres@localhost:5432/master`
+- `postgresql://postgres@localhost:5432/develop`
+- `postgresql://postgres@localhost:5432/chore_add_types`
+
+This is the same from CI (like github actions), except that instead of running against a local database, it is running against the supabase database. We will need a way inside our config to match branch names to database names too. For example 
+
+
+### Dump
+
+```sh
+supa db dump                # dumps the currently active (local) database to disk
+supa db restore             # restores the (local) database from disk
+supa migrate                # generate the migrations file
+supa up                     # runs the "up" migrations on Prod
+supa down                   # runs the "down" migrations on Prod
+```
+
+The `supa migrate` task is the trickiest part here. See [#migration-questions](#migration-questions) below.
+
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+- Supabase is about "time to value", and making things as simple/easy as possible. A CLI can be complicated, so we should make sure that the experience is as amazing as our hosted platform
+
+# Alternatives
+[alternatives]: #alternatives
+
+The hosted platform provides much of this functionality. We could continue to build this functionality into the platform however it will be a great dev experience if they can develop locally and push their changes later.
+
+# Unresolved questions
+[unresolved]: #unresolved-questions
+
+
+### Migration questions
+[migration-questions]: #migration-questions
+
+The `supa migrate` task is the trickiest part. How do we determine what migrations should run on our database, since the database is created declaratively? We would need to compare the databases somehow, and figure out the `up/down` migrations. Alternatively we could do "stateless" migrations, where we just run `restore` on a brand new database, then re-route traffic, but then how do we get the user's prod data into the new database. Bear in mind that enterprise users might have terrabytes of data, and so copying it across will be slow. 
+
+
+### Other questions
+- what is the prefix? should it be `supabase` (clearer) or `supa` (might have a few people calling it "supper/suppa")
+- Which CLI tool/framework/libraries should we use?
+- `supa start` requires docker-compose. Is it reasonable to expect that everyone will have docker set up on their computer? Can we make this easier? 
+- How much functionality should we push to postgres-meta? 
+
+
+
+# Future work
+[future]: #future-work
+
+### Supabase platform 
+
+```sh
+supa deploy   # Deploy a local app to prod. On the first time this is run, it should generate a `./supabase.js` with relevant Supabase config.
+supa orgs     # Manage orgs
+supa project  # Manage projects
+supa logs     # Fetch logs
+supa scale    # Scale the current database
+...etc
+```
+
+
+### Generators (bonus)
```

Comment:
This is a fantastic idea that I think will help a lot of people!

For some inspiration you can take a look at the fantastic [GraphQL Code Generator](https://graphql-code-generator.com/) project!